deployment/group_vars/all.yml

```diff
@@ -1,3 +1,34 @@
 ---
-injector_root_path: "/opt/3scale-perftest"
-injector_traffic_path: "{{ injector_root_path }}/traffic"
+
+RPS: Maximum requests per second to send
+DURATION: Duration of the performance test in seconds
+THREADS: Number of parallel threads to use
```

Comment:
To avoid confusion, that `THREADS` could be renamed to `CONCURRENCY` to refer to the number of shared connections. This is a key parameter in performance testing.  

Then make sure hyperfoil's `injector_hyperfoil_agent_threads` is at least `Concurrency` to ensure there is at least one hyperfoil `agent` assigned to each http connection.

Reply:
Let's not mix up terms here. An `agent` is an agent JVM running on the machine.
 In the commit already added I configured `injector_hyperfoil_agent_threads` = `THREADS`. To achieve what you've asked for.
 `THREADS` will be renamed to `CONCURRENCY`.

Reply:
JFYI, when performance tests are run the user wants to set RPS and number of concurrent connections opened against system under test. The user does not care about internal number of agent threads or anything like that which is specific of the testing tool like Hyperfoil. So, the exposed configuration to the user should be:

RPS: Maximum requests per second to send
DURATION: Duration of the performance test in seconds
CONCURRENCY: Number of parallel TCP connections opened

The ansible job is going to ensure the internal hyperfoil configuration to be consistent for the configured test in above terms. For instance, the  `injector_hyperfoil_agent_threads` attribute value is high enough to handle required load and so on..

Reply:
Fine, the value of `CONCURRENCY` will be assigned to `shared_connections`.